### PR TITLE
When calling RPC use cookies

### PR DESCRIPTION
**Pull request by dBucik**
- Prevents creating a new session in Perun for each RPC call
- Path to file with the cookie can be set from properties (property: perun.rpc.cookie.path)
  - the default value is /tmp/mitreid-rpc-cookie.txt

### CODE REVIEW

File: oidc-idp/src/main/java/cz/muni/ics/oidc/server/connectors/impl/PerunConnectorRpc.java [hunk 1]
```diff
@@ -97,11 +118,13 @@ public void postInit() {
...
 				.setConnectTimeout(30000) // Determines the timeout in milliseconds until a connection is established
 				.setSocketTimeout(60000) // The timeout for waiting for data
+				.setCookieSpec(CookieSpecs.DEFAULT)
 				.build();
+
 		PoolingHttpClientConnectionManager poolingConnectionManager = new PoolingHttpClientConnectionManager();
 		poolingConnectionManager.setMaxTotal(20); // maximum connections total
 		poolingConnectionManager.setDefaultMaxPerRoute(18);
-		ConnectionKeepAliveStrategy connectionKeepAliveStrategy = (response, context) -> {
+		ConnectionKeepAliveStrategy connectionKeepAliveStrategy = (response, clientContext) -> {
```
Comment (pajavyskocil) [inline oidc-idp/src/main/java/cz/muni/ics/oidc/server/connectors/impl/PerunConnectorRpc.java]: Why you are change 'context' to 'clientContext'

File: oidc-idp/src/main/java/cz/muni/ics/oidc/server/connectors/impl/PerunConnectorRpc.java [hunk 2]
```diff
@@ -479,4 +521,55 @@ private JsonNode makeRpcCall(String urlPart, Map<String, Object> map) {
 		}
 	}
 
+	private String extractPerunSessCookie(List<Cookie> cookies) {
```
Comment (pajavyskocil) [inline oidc-idp/src/main/java/cz/muni/ics/oidc/server/connectors/impl/PerunConnectorRpc.java]: Rename to 'extractPerunSessionCookie'

**Comment by dBucik** [comment]
Changed as requested.

**Comment by martin-kuba** [comment]
You can see the cookies going over the wire by enabling logging for
```
<logger name="org.apache.http.wire" level="trace" />
```

**Comment by dBucik** [comment]
Therefore, I am closing it. Thank you for pointing this fact out.

**Pull request closed without merging**